code/modules/reagents/machinery/dispenser/reagent_tank.dm

```diff
@@ -75,10 +112,63 @@
 		else
 	return
 
+/obj/structure/reagent_dispensers/bullet_act(var/obj/item/projectile/Proj)
+	if(Proj.get_structure_damage())
+		if(istype(Proj.firer))
+			message_admins("[key_name_admin(Proj.firer)] shot reagent tank at [loc.loc.name] ([loc.x],[loc.y],[loc.z]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[loc.x];Y=[loc.y];Z=[loc.z]'>JMP</a>).")
+			log_game("[key_name(Proj.firer)] shot reagent tank at [loc.loc.name] ([loc.x],[loc.y],[loc.z]).")
+
+		if(!istype(Proj ,/obj/item/projectile/beam/lasertag) && !istype(Proj ,/obj/item/projectile/beam/practice) )
```

Comment:
Are lasers sharp? I feel like sharpness is a good metric to check here, but if lasers aren't sharp, then I might suggest
```suggestion
		if(Proj.sharp || (istype(Proj, /obj/item/projectile/beam) && ![Proj.damage]))
```
That is, either the thing is sharp, or it is a beam that does damage (I think bean-bag rounds shouldn't rupture a tank either, as an example)